ganga/GangaCore/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -580,11 +580,7 @@ def run(self):
             # synchronize the main loop since we can get disable requests
             with self.__mainLoopCond:
 
-                log.debug("steps: %s" % str(self.steps))
-                log.debug("%s" % str(self.registry_slice))
-
-                log.debug("Monitoring loop __mainLoopCond")
-                log.debug('Monitoring loop lock acquired. Running loop')
+                log.debug('Monitoring loop lock acquired. Running loop: Steps %s' % str(self.steps))
```

Comment:
I can see that this is expensive for normal operation but if you would like to keep it only for the debug mode without impacting the normal operation you could consider changing this to the following.
```suggestion
                if log.isEnabledFor(logging.DEBUG):
                    log.debug('Monitoring loop lock acquired. Running loop: Steps %s' % str(self.steps))
```

Reply:
I considered this already but rejected for two reasons:

1. The statement is causing an occasional race condition where the list is looped on while it is added to. This is causing the monitoring thread to stop (and might actually be the cause of other trouble).
2. It is not even obvious that the debug statement was useful.

Reply:
fair enough